AutoMad/autoMad.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Oct 22 13:14:59 2024
+
+@author: dragon
+"""
+
+import random
+import re
+import wordGrabber
+import madLibFeeder
+import madDictionary
+import nltk
+import pandas as pd
+import numpy as np
+from nltk.corpus import stopwords
+from nltk.tokenize import word_tokenize, sent_tokenize
+
+NOUNS = ["NN","NNS"]
+VERBS = ["EX","MD","VB","VBD","VBG","VBN","VBP","VBZ"]
+CONJUNCTIONS =["CC"]
+NUMBER = ["CD","LS"]
+ADJECTIVES = ["JJ","JJR","JJS","WDT"]
+ADVERBS = ["RB","RBR","RBS","RP","WRB"]
+SILLIES = ["UH"]
+UNKNOWN = ["FW","PDT","POS"]
+PREPOSITIONS = ["IN","TO","WP","WP$"]
+PRONOUNS = ["PRP","PRP$"]
+ARTICLES = ["DT"]
+NAMES = ["NNP","NNPS"]
+
+def posToCategory(pos):
```

Comment:
Anywhere I see a long chain of `if`s I start looking for ways to do it with a dictionary instead. Here I think you could do something like

```
PARTS_OF_SPEECH = {
  'noun' = NOUNS,
  'verb' = VERBS'
  # ...
def posToGategory(pos):
  for cat, posList in PARTS_OF_SPEECH.items():
    if pos in posList:
       return cat
```

Reply:
or
```
PARTS_OF_SPEECH = {
  'noun': ["NN","NNS"]
  # ...
}